### PR TITLE
Fixing bugs in java format and added readValidator

### PR DESCRIPTION
**Pull request by saleto**
I added a few change according to what we discussed, please review my code and merge @hoan9it

### CODE REVIEW

File: src/main/java/pb360/service/impl/GenerateRestApiServiceImpl.java [hunk 1]
```diff
@@ -244,13 +260,41 @@ private void writeToFinalFile(JsonNode jsonNode, String name, String controllerN
...
-//		input = input.replace("null", "");
+	
 		
-		input = input.replace("{1}", controllerName);
-		input = input.replace("ControllerName", controllerName);
-		writeToJavaFile(input, name);
+//		input = input.replace("ControllerName", controllerName);
+//		if(fileIndex == 0)
+//		{
+//			writeToJavaFile(input, name, "D:\\restapi\\nextgen-sti\\src\\main\\java\\pb360\\model");
```
Comment (hoan-salt) [inline src/main/java/pb360/service/impl/GenerateRestApiServiceImpl.java]: don't use hard code, please remove. please research about use constant for static variable in java

File: src/main/java/pb360/data/entity/RestApi.java [hunk 2]
```diff
@@ -29,10 +29,10 @@ public RestApi() {
 		super();
 	}
 
-	public RestApi(@JsonProperty("restId") String restId, @JsonProperty("restName") String restName,
-			@JsonProperty("restUrl") String restUrl, @JsonProperty("restStatus") MessageObject restStatus,
-			@JsonProperty("fileOfRest") List<String> fileOfRest, @JsonProperty("restLocation") String restLocation,
-			@JsonProperty("lastModifying") Date lastModifying) {
+	public RestApi( String restId,  String restName,
```
Comment (hoan-salt) [inline src/main/java/pb360/data/entity/RestApi.java:32]: why we need to remove JsonProperty on the constructor

File: src/main/java/pb360/data/repository/MessageObjectRepository.java [hunk 3]
```diff
@@ -1,9 +1,18 @@
...
 
+import pb360.data.entity.RestApi;
 import pb360.model.MessageObject;
 
-public interface MessageObjectRepository extends MongoRepository<MessageObject, String> {
+public interface MessageObjectRepository extends MongoRepository<RestApi, String> {
+	
+//	public RestApi findBymessageNumber(String messageNumber);
+//	public RestApi findFirstByOrderByrestIdDesc();
+	public int countByrestId();
```
Comment (hoan-salt) [inline src/main/java/pb360/data/repository/MessageObjectRepository.java:14]: change return type to Integer, sometime you will get null when DB is empty, int cannot contain a null value

File: src/main/java/pb360/data/repository/RestApiRepository.java [hunk 4]
```diff
@@ -4,5 +4,5 @@
 import pb360.data.entity.RestApi;
 
 public interface RestApiRepository extends MongoRepository<RestApi, String> {
-
+	int countByrestId();
```
Comment (hoan-salt) [inline src/main/java/pb360/data/repository/RestApiRepository.java:7]: please correct

File: src/main/java/pb360/model/MessageObject.java [hunk 5]
```diff
@@ -20,7 +23,7 @@ public MessageObject() {
 	}
 
 	public MessageObject(@JsonProperty("datetime") Date datetime, @JsonProperty("type") String type,
-			@JsonProperty("data") String data) {
+			@JsonProperty("data") String data, @JsonProperty("_id")String messageNumber) {
```
Comment (hoan-salt) [inline src/main/java/pb360/model/MessageObject.java:26]: the property should be same as variable

File: src/main/java/pb360/service/impl/GenerateRestApiServiceImpl.java [hunk 6]
```diff
@@ -37,8 +60,22 @@ public MessageObject getRestApiData(String restId) {
 		getMessage.setType("GET");
 		getMessage.setDatetime(getCurrentDateTime().getDatetime());
 
+		@SuppressWarnings("resource")
```
Comment (hoan-salt) [inline src/main/java/pb360/service/impl/GenerateRestApiServiceImpl.java:63]: what the resource? what are you doing on this code, we have a application.properties to provide this configuration, why you need to add more configuration at here? please contact Phuoc about how to read data from DB, do not copy a code from the internet then parse to project if you don't understand

File: src/main/java/pb360/service/impl/GenerateRestApiServiceImpl.java [hunk 7]
```diff
@@ -56,36 +93,50 @@ public MessageObject getRestApiData(String restId) {
 	@Override
 	public MessageObject createRestApiData(RestAPI restApi) {
 		List<String> fileOfRest = new ArrayList<>();
-		
-		restApi.setRestId(_id+"");
-		_id = _id ++;
+		RestApi entity = new RestApi();
+
+		@SuppressWarnings("resource")
+		MongoClient mongoClient = new MongoClient("localhost", 27017);
```
Comment (hoan-salt) [inline src/main/java/pb360/service/impl/GenerateRestApiServiceImpl.java:99]: remove, contact me if you didn't understand what are you doing

File: src/main/java/pb360/service/impl/GenerateRestApiServiceImpl.java [hunk 8]
```diff
@@ -105,6 +156,21 @@ public MessageObject deleteRestApiData(String restId) {
 		delete.setType("DELETE");
 		delete.setDatetime(getCurrentDateTime().getDatetime());
 
+		@SuppressWarnings("resource")
+		MongoClient mongoClient = new MongoClient("localhost", 27017);
+		MongoDatabase database = mongoClient.getDatabase("restapidbs");
+		MongoCollection<Document> collection = database.getCollection("RestApi");
```
Comment (hoan-salt) [inline src/main/java/pb360/service/impl/GenerateRestApiServiceImpl.java:162]: I don't know what are you doing Sang

**Comment by hoan-salt** [comment]
Please resolve conflict

File: src/main/java/pb360/service/impl/GenerateRestApiServiceImpl.java (hunk 6 above)
Comment (saleto) [reply src/main/java/pb360/service/impl/GenerateRestApiServiceImpl.java:63]: This was added from suggestion by Eclipse (Resource leak: 'mongoClient' is never closed)

File: src/main/java/pb360/service/impl/GenerateRestApiServiceImpl.java (hunk 7 above)
Comment (saleto) [reply src/main/java/pb360/service/impl/GenerateRestApiServiceImpl.java:99]: @SuppressWarnings("resource") will be removed

File: src/main/java/pb360/service/impl/GenerateRestApiServiceImpl.java (hunk 8 above)
Comment (saleto) [reply src/main/java/pb360/service/impl/GenerateRestApiServiceImpl.java:162]: This is another way to read document/collection from database.
mongoClient to tell the address
database to get database's name
collection to point to RestApi collection

And from collection i can count/find a field in documents

**Pull request closed without merging**